db/postgres.go

```diff
@@ -221,6 +224,51 @@ func (p *pgClient) WriteTodaysUsage(counts map[string]api.RelayCounts) error {
 	return nil
 }
 
+// WriteTodaysUsage writes the app metrics for today so far to the underlying PG table.
+//	All the entries in the table holding todays metrics are deleted first.
+func (p *pgClient) WriteTodaysLatencies(latencies map[string][]api.Latency) error {
+	ctx := context.Background()
+	// TODO: determine required isolation level
+	tx, err := p.DB.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
+	if err != nil {
+		return err
+	}
+
+	// todays_app_latencies table gets rebuilt every time
+	_, deleteErr := tx.ExecContext(ctx, "DELETE FROM todays_app_latencies")
+	if deleteErr != nil {
+		if rollbackErr := tx.Rollback(); rollbackErr != nil {
+			fmt.Printf("delete failed: %v, unable to rollback: %v\n", deleteErr, rollbackErr)
+			return deleteErr
+		}
+	}
+
+	// TODO: bulk insert
+	for app, latencies := range latencies {
+
+		latenciesString, err := json.Marshal(latencies)
```

Comment:
We are writing this to a postgres DB. This is trying to store multiple value in a single column, which I think should be avoided. Any other solutions, e.g. multiple rows per application (preferable) or a set number of columns to store a limited number of relays per application, works for me (as long as it conforms to relational data model)

Reply:
Updated to store each latency result to a single row in the table